### ISSUE TITLE
support JSON text sequences in stream_in/_out

### ISSUE THREAD

**Issue by sckott** [jeroen/jsonlite#171]
@jeroenooms we talked about this briefly - opening an issue to talk about this in case you think it can be supported alongside ndjson

links: 

* JSON Text Sequences specification: https://tools.ietf.org/html/rfc7464, brief description from it:

> A JSON text sequence consists of any number of JSON texts, all encoded in UTF-8, each prefixed by an ASCII Record Separator (0x1E), and each ending with an ASCII Line Feed character (0x0A)

* JSON text sequences are supported in `jq` 
    * parses json text sequence format, e.g., https://github.com/stedolan/jq/issues/597 
    * see `--seq` flag in https://stedolan.github.io/jq/manual/#Invokingjq

* some python egs work with `rfc7464` library

```bash
hub clone moshez/rfc7464
cd rfc7464
python setup.py build && python setup.py install
```

```python
>>> import rfc7464
```

```python
>>> rfc7464.emit(1)
b'\x1e1\n'
>>> rfc7464.emit(u"hello")
b'\x1e"hello"\n'
>>> rfc7464.emit([1, 2, 3])
b'\x1e[1, 2, 3]\n'
>>> rfc7464.emit({u'hello': 5, u'world': 6})
```

```python
>>> p = rfc7464.Parser()
>>> inp = (b'\x1e' + json.dumps([1]).encode('utf-8') + b'\n' +
            b'\x1e' + json.dumps({u'hello': 5}).encode('utf-8') + b'\n')
>>> list(p.receive(inp))
[[1], {'hello': 5}]
```

I want this in general, but more specifically for GeoJSON text sequences (a proposed standard on top of JSON text sequences) https://github.com/geojson/geojson-text-sequences  so we can support read/write of geojson text sequences in our various R clients - supports use cases where users want to process line by line instead of read all then process - Logic specific to GeoJSON could go into our `geojson` package e.g. 

if it matters here, the media type is `application/json-seq`, while ndjson is `application/x-ndjson`

**Comment by jeroenooms** [comment]
Do we have some real size example `JSON text sequence` data to test with?

**Comment by sckott** [comment]
I'll get some

**Comment by jeroenooms** [comment]
I added some experimental support:

```r
tmp <- tempfile()
stream_out(iris, file(tmp), prefix = "\u001E")
stream_in(file(tmp))
```

But I'm not convinced this is the way to go.

**Comment by jeroen** [comment]
I think this is good. Has landed in jsonlite v1.4
